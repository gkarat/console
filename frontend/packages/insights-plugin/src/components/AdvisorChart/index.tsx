import * as React from 'react';
import { useTranslation } from 'react-i18next';
import { ChartDonut, ChartLabel, ChartLegend } from '@patternfly/react-charts';
import { global_Color_dark_200 as globalColorDark200 } from '@patternfly/react-tokens';

import {
  AdvisorChartTitle,
  AdvisorChartLegendIcon,
  chartColorScale,
  riskLabels,
} from './chartHelpers';

const AdvisorChart = ({ metrics, clusterId }) => {
  const { t } = useTranslation();

  const issues = Object.entries(metrics);
  const issuesCount = Object.values(metrics).reduce<number>(
    (acc: number, cur: number) => acc + cur,
    0,
  );

  return (
    <ChartDonut
      data={issues.map(([k, v]) => ({
        label: `${v} ${t(riskLabels[k])}`,
        x: k,
        y: v,
      }))}
      title={`${issuesCount}`}
      subTitle={
        issuesCount === 1
          ? t('insights-plugin~Total issue', { count: issuesCount })
          : t('insights-plugin~Total issues', { count: issuesCount })
      }
      subTitleComponent={<ChartLabel style={{ marginTop: 5, fill: globalColorDark200.value }} />}
      legendData={Object.entries(metrics).map(([k, v]) => ({ name: `${k}: ${v}` }))}
      width={300}
      height={150}
      colorScale={chartColorScale}
      legendComponent={
        <ChartLegend
          data={issues.map(([k, v]) => ({
            name: `${v} ${t(riskLabels[k])}`,
            id: k,
          }))}
          labelComponent={<AdvisorChartTitle clusterId={clusterId} />}
          dataComponent={<AdvisorChartLegendIcon />}
          title={t('insights-plugin~Total Risk')}
          titleComponent={<ChartLabel style={{ fontWeight: 'bold' }} />}
          x={0}
          orientation="vertical"
          rowGutter={-5}
        />
      }
      padding={{
        bottom: 0,
        left: 150,
        right: 0, // Adjusted to accommodate legend
        top: 0,
      }}
    />
  );
};

export default AdvisorChart;
