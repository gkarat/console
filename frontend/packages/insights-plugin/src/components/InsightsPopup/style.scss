.co-insights__no-rules {
  color: var(--pf-global--Color--200);
}

.insights-advisor--enabled-link {
  tspan {
    fill: var(
      --pf-global--link--Color
    ) !important; // use !important to override inline style applied by Victory charts

    &:hover {
      fill: var(--pf-global--link--Color--hover) !important;
    }
  }
}
