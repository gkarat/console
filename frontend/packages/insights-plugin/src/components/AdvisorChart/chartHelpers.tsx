import * as React from 'react';
import { ChartLabel } from '@patternfly/react-charts';
import { AngleDoubleDownIcon, AngleDoubleUpIcon, EqualsIcon } from '@patternfly/react-icons';

import { global_palette_blue_50 as blue50 } from '@patternfly/react-tokens/dist/js/global_palette_blue_50';
import { global_palette_blue_300 as blue300 } from '@patternfly/react-tokens/dist/js/global_palette_blue_300';
import { global_palette_gold_400 as gold400 } from '@patternfly/react-tokens/dist/js/global_palette_gold_400';
import { global_palette_orange_300 as orange300 } from '@patternfly/react-tokens/dist/js/global_palette_orange_300';
import { global_palette_red_200 as red200 } from '@patternfly/react-tokens/dist/js/global_palette_red_200';

import CriticalIcon from '../CriticalIcon';
import { ExternalLink } from '@console/internal/components/utils';

const chartColorScale = [blue50.value, gold400.value, orange300.value, red200.value];

const legendColorScale = {
  low: blue300.value,
  moderate: gold400.value,
  important: orange300.value,
  critical: red200.value,
};

const riskIcons = {
  low: AngleDoubleDownIcon,
  moderate: EqualsIcon,
  important: AngleDoubleUpIcon,
  critical: CriticalIcon,
};

const riskLabels = {
  // t('insights-plugin~Critical')
  critical: 'insights-plugin~Critical',
  // t('insights-plugin~Important')
  important: 'insights-plugin~Important',
  // t('insights-plugin~Moderate')
  moderate: 'insights-plugin~Moderate',
  // t('insights-plugin~Low')
  low: 'insights-plugin~Low',
};

const AdvisorChartTitle = (props) => {
  const { clusterId } = props;

  return clusterId ? (
    <ExternalLink href={`https://cloud.redhat.com/openshift/details/${clusterId}#insights`}>
      <ChartLabel
        {...props}
        style={[{ fontSize: 13 }]}
        className="insights-advisor--enabled-link"
      />
    </ExternalLink>
  ) : (
    <ChartLabel {...props} style={[{ fontSize: 13 }]} />
  );
};

const AdvisorChartLegendIcon = ({ x, y, datum }) => {
  const Icon = riskIcons[datum.id];
  return <Icon x={x - 3} y={y - 6} fill={legendColorScale[datum.id]} />;
};

export { chartColorScale, riskLabels, AdvisorChartTitle, AdvisorChartLegendIcon };
