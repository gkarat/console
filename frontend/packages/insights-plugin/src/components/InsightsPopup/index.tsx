import * as React from 'react';
import { useTranslation } from 'react-i18next';
import { PrometheusHealthPopupProps } from '@console/plugin-sdk';
import { K8sResourceKind } from '@console/internal/module/k8s';
import { ExternalLink, openshiftHelpBase } from '@console/internal/components/utils';

import AdvisorChart from '../AdvisorChart';
import {
  mapMetrics,
  isWaitingOrDisabled as _isWaitingOrDisabled,
  isError as _isError,
} from '../../mappers';
import './style.scss';

export const InsightsPopup: React.FC<PrometheusHealthPopupProps> = ({ responses, k8sResult }) => {
  const { t } = useTranslation();
  const metrics = mapMetrics(responses[0].response);
  const clusterId = (k8sResult as K8sResourceKind)?.data?.spec?.clusterId || '';
  const isWaitingOrDisabled = _isWaitingOrDisabled(metrics);
  const isError = _isError(metrics);

  return (
    <div className="co-insights__box">
      <p id="insights-plugin--header">
        {t(
          'insights-plugin~Insights Advisor identifies and prioritizes risks to security, performance, availability, and stability of your clusters.',
        )}
      </p>
      {isError && (
        <div className="co-status-popup__section text-secondary">
          {t('insights-plugin~Temporary unavailable.')}
        </div>
      )}
      {isWaitingOrDisabled && (
        <div className="co-status-popup__section text-secondary">
          {t('insights-plugin~Disabled or waiting for results.')}
        </div>
      )}
      {!isWaitingOrDisabled && !isError && <AdvisorChart metrics={metrics} clusterId={clusterId} />}
      <div className="co-status-popup__section">
        {!isWaitingOrDisabled && !isError && clusterId && (
          <>
            <h6 className="pf-c-title pf-m-md">{t('insights-plugin~Fixable issues')}</h6>
            <ExternalLink
              href={`https://cloud.redhat.com/openshift/details/${clusterId}#insights`}
              text={t('insights-plugin~View all in OpenShift Cluster Manager')}
            />
          </>
        )}
        {!isWaitingOrDisabled && !isError && !clusterId && (
          <ExternalLink
            href={`https://cloud.redhat.com/openshift/`}
            text={t('insights-plugin~Go to OpenShift Cluster Manager')}
          />
        )}
        {(isWaitingOrDisabled || isError) && (
          <ExternalLink
            href={`${openshiftHelpBase}support/remote_health_monitoring/using-insights-to-identify-issues-with-your-cluster.html`}
            text={t('insights-plugin~More about Insights')}
          />
        )}
      </div>
    </div>
  );
};

InsightsPopup.displayName = 'InsightsPopup';
