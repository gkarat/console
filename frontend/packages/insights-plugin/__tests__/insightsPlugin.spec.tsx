import * as React from 'react';
import { shallow } from 'enzyme';

import { InsightsPopup } from '../src/components/InsightsPopup';
import AdvisorChart from '../src/components/AdvisorChart';
import fixtures from './insightsPlugin.fixtures';

describe('Insights plugin', () => {
  describe('<InsightsPopup /> OK state', () => {
    const wrapper = shallow(<InsightsPopup {...fixtures.propsInsightsPopup} />);

    it('should render', () => {
      expect(wrapper).toMatchSnapshot();
    });
    it('renders main content component', () => {
      expect(wrapper.find('.co-insights__box').length).toBe(1);
    });
    it('should have header', () => {
      expect(wrapper.find('#insights-plugin--header').text()).toBe(
        'Insights Advisor identifies and prioritizes risks to security, performance, availability, and stability of your clusters.',
      );
    });
    it('should render chart', () => {
      expect(wrapper.find(AdvisorChart).length).toBe(1);
    });
  });

  describe('<InsightsPopup /> error state', () => {
    const { responses } = fixtures.propsInsightsPopup;
    responses[0].response.data.result = [];
    const props = {
      ...fixtures,
      responses,
    };
    const wrapper = shallow(<InsightsPopup {...props} />);

    it('should render', () => {
      expect(wrapper).toMatchSnapshot();
    });
    it('renders main content component', () => {
      expect(wrapper.find('.co-insights__box').length).toBe(1);
    });
    it('should have header', () => {
      expect(wrapper.find('#insights-plugin--header').text()).toBe(
        'Insights Advisor identifies and prioritizes risks to security, performance, availability, and stability of your clusters.',
      );
    });
    it('should have error text message', () => {
      expect(wrapper.find('.co-status-popup__section.text-secondary').text()).toBe(
        'Temporary unavailable.',
      );
    });
  });

  describe('<InsightsPopup /> disabled state', () => {
    const wrapper = shallow(<InsightsPopup {...fixtures.propsInsightsPopupDisabled} />);

    it('should render', () => {
      expect(wrapper).toMatchSnapshot();
    });
    it('renders main content component', () => {
      expect(wrapper.find('.co-insights__box').length).toBe(1);
    });
    it('should have header', () => {
      expect(wrapper.find('#insights-plugin--header').text()).toBe(
        'Insights Advisor identifies and prioritizes risks to security, performance, availability, and stability of your clusters.',
      );
    });
    it('should have error text message', () => {
      expect(wrapper.find('.co-status-popup__section.text-secondary').text()).toBe(
        'Disabled or waiting for results.',
      );
    });
  });
});
