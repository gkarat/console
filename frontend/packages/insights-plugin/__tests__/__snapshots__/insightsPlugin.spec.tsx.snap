// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Insights plugin <InsightsPopup /> OK state should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InsightsPopup
    k8sResult={
      Object {
        "data": Object {
          "spec": Object {
            "clusterID": "d4b48e95-83e8-436c-b2c9-283c05aecc1c",
          },
        },
        "loadError": "",
        "loaded": true,
      }
    }
    responses={
      Array [
        Object {
          "error": null,
          "response": Object {
            "data": Object {
              "result": Array [],
              "resultType": "vector",
            },
            "status": "success",
          },
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          id="insights-plugin--header"
        >
          Insights Advisor identifies and prioritizes risks to security, performance, availability, and stability of your clusters.
        </p>,
        false,
        false,
        <AdvisorChart
          clusterId=""
          metrics={
            Object {
              "critical": 0,
              "important": 0,
              "low": 0,
              "moderate": 1,
            }
          }
        />,
        <div
          className="co-status-popup__section"
        >
          
          <Unknown
            href="https://cloud.redhat.com/openshift/"
            text="Go to OpenShift Cluster Manager"
          />
        </div>,
      ],
      "className": "co-insights__box",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Insights Advisor identifies and prioritizes risks to security, performance, availability, and stability of your clusters.",
          "id": "insights-plugin--header",
        },
        "ref": null,
        "rendered": "Insights Advisor identifies and prioritizes risks to security, performance, availability, and stability of your clusters.",
        "type": "p",
      },
      false,
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "clusterId": "",
          "metrics": Object {
            "critical": 0,
            "important": 0,
            "low": 0,
            "moderate": 1,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "",
            <Unknown
              href="https://cloud.redhat.com/openshift/"
              text="Go to OpenShift Cluster Manager"
            />,
            false,
          ],
          "className": "co-status-popup__section",
        },
        "ref": null,
        "rendered": Array [
          "",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "href": "https://cloud.redhat.com/openshift/",
              "text": "Go to OpenShift Cluster Manager",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          false,
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            id="insights-plugin--header"
          >
            Insights Advisor identifies and prioritizes risks to security, performance, availability, and stability of your clusters.
          </p>,
          false,
          false,
          <AdvisorChart
            clusterId=""
            metrics={
              Object {
                "critical": 0,
                "important": 0,
                "low": 0,
                "moderate": 1,
              }
            }
          />,
          <div
            className="co-status-popup__section"
          >
            
            <Unknown
              href="https://cloud.redhat.com/openshift/"
              text="Go to OpenShift Cluster Manager"
            />
          </div>,
        ],
        "className": "co-insights__box",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Insights Advisor identifies and prioritizes risks to security, performance, availability, and stability of your clusters.",
            "id": "insights-plugin--header",
          },
          "ref": null,
          "rendered": "Insights Advisor identifies and prioritizes risks to security, performance, availability, and stability of your clusters.",
          "type": "p",
        },
        false,
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "clusterId": "",
            "metrics": Object {
              "critical": 0,
              "important": 0,
              "low": 0,
              "moderate": 1,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "",
              <Unknown
                href="https://cloud.redhat.com/openshift/"
                text="Go to OpenShift Cluster Manager"
              />,
              false,
            ],
            "className": "co-status-popup__section",
          },
          "ref": null,
          "rendered": Array [
            "",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "href": "https://cloud.redhat.com/openshift/",
                "text": "Go to OpenShift Cluster Manager",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            false,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSeventeenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`Insights plugin <InsightsPopup /> disabled state should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InsightsPopup
    k8sResult={
      Object {
        "data": Object {
          "spec": Object {
            "clusterID": "d4b48e95-83e8-436c-b2c9-283c05aecc1c",
          },
        },
        "loadError": "",
        "loaded": true,
      }
    }
    responses={
      Array [
        Object {
          "error": null,
          "response": Object {
            "data": Object {
              "result": Array [
                Object {
                  "metric": Object {
                    "__name__": "health_statuses_insights",
                    "container": "insights-operator",
                    "endpoint": "https",
                    "instance": "10.128.0.5:8443",
                    "job": "metrics",
                    "metric": "critical",
                    "namespace": "openshift-insights",
                    "pod": "insights-operator-7b5d69cd76-2nkvj",
                    "prometheus": "openshift-monitoring/k8s",
                    "service": "metrics",
                  },
                  "value": Array [
                    1622715672.165,
                    "-1",
                  ],
                },
                Object {
                  "metric": Object {
                    "__name__": "health_statuses_insights",
                    "container": "insights-operator",
                    "endpoint": "https",
                    "instance": "10.128.0.5:8443",
                    "job": "metrics",
                    "metric": "important",
                    "namespace": "openshift-insights",
                    "pod": "insights-operator-7b5d69cd76-2nkvj",
                    "prometheus": "openshift-monitoring/k8s",
                    "service": "metrics",
                  },
                  "value": Array [
                    1622715672.165,
                    "-1",
                  ],
                },
                Object {
                  "metric": Object {
                    "__name__": "health_statuses_insights",
                    "container": "insights-operator",
                    "endpoint": "https",
                    "instance": "10.128.0.5:8443",
                    "job": "metrics",
                    "metric": "low",
                    "namespace": "openshift-insights",
                    "pod": "insights-operator-7b5d69cd76-2nkvj",
                    "prometheus": "openshift-monitoring/k8s",
                    "service": "metrics",
                  },
                  "value": Array [
                    1622715672.165,
                    "-1",
                  ],
                },
                Object {
                  "metric": Object {
                    "__name__": "health_statuses_insights",
                    "container": "insights-operator",
                    "endpoint": "https",
                    "instance": "10.128.0.5:8443",
                    "job": "metrics",
                    "metric": "moderate",
                    "namespace": "openshift-insights",
                    "pod": "insights-operator-7b5d69cd76-2nkvj",
                    "prometheus": "openshift-monitoring/k8s",
                    "service": "metrics",
                  },
                  "value": Array [
                    1622715672.165,
                    "-1",
                  ],
                },
              ],
              "resultType": "vector",
            },
            "status": "success",
          },
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          id="insights-plugin--header"
        >
          Insights Advisor identifies and prioritizes risks to security, performance, availability, and stability of your clusters.
        </p>,
        false,
        <div
          className="co-status-popup__section text-secondary"
        >
          Disabled or waiting for results.
        </div>,
        false,
        <div
          className="co-status-popup__section"
        >
          <Unknown
            href="https://docs.okd.io/latest/support/remote_health_monitoring/using-insights-to-identify-issues-with-your-cluster.html"
            text="More about Insights"
          />
        </div>,
      ],
      "className": "co-insights__box",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Insights Advisor identifies and prioritizes risks to security, performance, availability, and stability of your clusters.",
          "id": "insights-plugin--header",
        },
        "ref": null,
        "rendered": "Insights Advisor identifies and prioritizes risks to security, performance, availability, and stability of your clusters.",
        "type": "p",
      },
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Disabled or waiting for results.",
          "className": "co-status-popup__section text-secondary",
        },
        "ref": null,
        "rendered": "Disabled or waiting for results.",
        "type": "div",
      },
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            false,
            false,
            <Unknown
              href="https://docs.okd.io/latest/support/remote_health_monitoring/using-insights-to-identify-issues-with-your-cluster.html"
              text="More about Insights"
            />,
          ],
          "className": "co-status-popup__section",
        },
        "ref": null,
        "rendered": Array [
          false,
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "href": "https://docs.okd.io/latest/support/remote_health_monitoring/using-insights-to-identify-issues-with-your-cluster.html",
              "text": "More about Insights",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            id="insights-plugin--header"
          >
            Insights Advisor identifies and prioritizes risks to security, performance, availability, and stability of your clusters.
          </p>,
          false,
          <div
            className="co-status-popup__section text-secondary"
          >
            Disabled or waiting for results.
          </div>,
          false,
          <div
            className="co-status-popup__section"
          >
            <Unknown
              href="https://docs.okd.io/latest/support/remote_health_monitoring/using-insights-to-identify-issues-with-your-cluster.html"
              text="More about Insights"
            />
          </div>,
        ],
        "className": "co-insights__box",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Insights Advisor identifies and prioritizes risks to security, performance, availability, and stability of your clusters.",
            "id": "insights-plugin--header",
          },
          "ref": null,
          "rendered": "Insights Advisor identifies and prioritizes risks to security, performance, availability, and stability of your clusters.",
          "type": "p",
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Disabled or waiting for results.",
            "className": "co-status-popup__section text-secondary",
          },
          "ref": null,
          "rendered": "Disabled or waiting for results.",
          "type": "div",
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              false,
              false,
              <Unknown
                href="https://docs.okd.io/latest/support/remote_health_monitoring/using-insights-to-identify-issues-with-your-cluster.html"
                text="More about Insights"
              />,
            ],
            "className": "co-status-popup__section",
          },
          "ref": null,
          "rendered": Array [
            false,
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "href": "https://docs.okd.io/latest/support/remote_health_monitoring/using-insights-to-identify-issues-with-your-cluster.html",
                "text": "More about Insights",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSeventeenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`Insights plugin <InsightsPopup /> error state should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InsightsPopup
    propsInsightsPopup={
      Object {
        "k8sResult": Object {
          "data": Object {
            "spec": Object {
              "clusterID": "d4b48e95-83e8-436c-b2c9-283c05aecc1c",
            },
          },
          "loadError": "",
          "loaded": true,
        },
        "responses": Array [
          Object {
            "error": null,
            "response": Object {
              "data": Object {
                "result": Array [],
                "resultType": "vector",
              },
              "status": "success",
            },
          },
        ],
      }
    }
    propsInsightsPopupDisabled={
      Object {
        "k8sResult": Object {
          "data": Object {
            "spec": Object {
              "clusterID": "d4b48e95-83e8-436c-b2c9-283c05aecc1c",
            },
          },
          "loadError": "",
          "loaded": true,
        },
        "responses": Array [
          Object {
            "error": null,
            "response": Object {
              "data": Object {
                "result": Array [
                  Object {
                    "metric": Object {
                      "__name__": "health_statuses_insights",
                      "container": "insights-operator",
                      "endpoint": "https",
                      "instance": "10.128.0.5:8443",
                      "job": "metrics",
                      "metric": "critical",
                      "namespace": "openshift-insights",
                      "pod": "insights-operator-7b5d69cd76-2nkvj",
                      "prometheus": "openshift-monitoring/k8s",
                      "service": "metrics",
                    },
                    "value": Array [
                      1622715672.165,
                      "-1",
                    ],
                  },
                  Object {
                    "metric": Object {
                      "__name__": "health_statuses_insights",
                      "container": "insights-operator",
                      "endpoint": "https",
                      "instance": "10.128.0.5:8443",
                      "job": "metrics",
                      "metric": "important",
                      "namespace": "openshift-insights",
                      "pod": "insights-operator-7b5d69cd76-2nkvj",
                      "prometheus": "openshift-monitoring/k8s",
                      "service": "metrics",
                    },
                    "value": Array [
                      1622715672.165,
                      "-1",
                    ],
                  },
                  Object {
                    "metric": Object {
                      "__name__": "health_statuses_insights",
                      "container": "insights-operator",
                      "endpoint": "https",
                      "instance": "10.128.0.5:8443",
                      "job": "metrics",
                      "metric": "low",
                      "namespace": "openshift-insights",
                      "pod": "insights-operator-7b5d69cd76-2nkvj",
                      "prometheus": "openshift-monitoring/k8s",
                      "service": "metrics",
                    },
                    "value": Array [
                      1622715672.165,
                      "-1",
                    ],
                  },
                  Object {
                    "metric": Object {
                      "__name__": "health_statuses_insights",
                      "container": "insights-operator",
                      "endpoint": "https",
                      "instance": "10.128.0.5:8443",
                      "job": "metrics",
                      "metric": "moderate",
                      "namespace": "openshift-insights",
                      "pod": "insights-operator-7b5d69cd76-2nkvj",
                      "prometheus": "openshift-monitoring/k8s",
                      "service": "metrics",
                    },
                    "value": Array [
                      1622715672.165,
                      "-1",
                    ],
                  },
                ],
                "resultType": "vector",
              },
              "status": "success",
            },
          },
        ],
      }
    }
    responses={
      Array [
        Object {
          "error": null,
          "response": Object {
            "data": Object {
              "result": Array [],
              "resultType": "vector",
            },
            "status": "success",
          },
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          id="insights-plugin--header"
        >
          Insights Advisor identifies and prioritizes risks to security, performance, availability, and stability of your clusters.
        </p>,
        <div
          className="co-status-popup__section text-secondary"
        >
          Temporary unavailable.
        </div>,
        false,
        false,
        <div
          className="co-status-popup__section"
        >
          <Unknown
            href="https://docs.okd.io/latest/support/remote_health_monitoring/using-insights-to-identify-issues-with-your-cluster.html"
            text="More about Insights"
          />
        </div>,
      ],
      "className": "co-insights__box",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Insights Advisor identifies and prioritizes risks to security, performance, availability, and stability of your clusters.",
          "id": "insights-plugin--header",
        },
        "ref": null,
        "rendered": "Insights Advisor identifies and prioritizes risks to security, performance, availability, and stability of your clusters.",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Temporary unavailable.",
          "className": "co-status-popup__section text-secondary",
        },
        "ref": null,
        "rendered": "Temporary unavailable.",
        "type": "div",
      },
      false,
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            false,
            false,
            <Unknown
              href="https://docs.okd.io/latest/support/remote_health_monitoring/using-insights-to-identify-issues-with-your-cluster.html"
              text="More about Insights"
            />,
          ],
          "className": "co-status-popup__section",
        },
        "ref": null,
        "rendered": Array [
          false,
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "href": "https://docs.okd.io/latest/support/remote_health_monitoring/using-insights-to-identify-issues-with-your-cluster.html",
              "text": "More about Insights",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            id="insights-plugin--header"
          >
            Insights Advisor identifies and prioritizes risks to security, performance, availability, and stability of your clusters.
          </p>,
          <div
            className="co-status-popup__section text-secondary"
          >
            Temporary unavailable.
          </div>,
          false,
          false,
          <div
            className="co-status-popup__section"
          >
            <Unknown
              href="https://docs.okd.io/latest/support/remote_health_monitoring/using-insights-to-identify-issues-with-your-cluster.html"
              text="More about Insights"
            />
          </div>,
        ],
        "className": "co-insights__box",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Insights Advisor identifies and prioritizes risks to security, performance, availability, and stability of your clusters.",
            "id": "insights-plugin--header",
          },
          "ref": null,
          "rendered": "Insights Advisor identifies and prioritizes risks to security, performance, availability, and stability of your clusters.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Temporary unavailable.",
            "className": "co-status-popup__section text-secondary",
          },
          "ref": null,
          "rendered": "Temporary unavailable.",
          "type": "div",
        },
        false,
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              false,
              false,
              <Unknown
                href="https://docs.okd.io/latest/support/remote_health_monitoring/using-insights-to-identify-issues-with-your-cluster.html"
                text="More about Insights"
              />,
            ],
            "className": "co-status-popup__section",
          },
          "ref": null,
          "rendered": Array [
            false,
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "href": "https://docs.okd.io/latest/support/remote_health_monitoring/using-insights-to-identify-issues-with-your-cluster.html",
                "text": "More about Insights",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSeventeenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
